fenicsadapter/fenicsadapter.py

```diff
@@ -299,458 +73,236 @@ def __init__(self, adapter_config_filename='precice-adapter-config.json',
         except (TypeError, KeyError):  # works with dolfin 2017.2.0
-            self._coupling_bc_expression = self._my_expression(element=self._function_space.ufl_element(), degree=0)
-        self._coupling_bc_expression.set_boundary_data(self._read_data, x_vert, y_vert)
-
-    def create_coupling_dirichlet_boundary_condition(self, function_space):
-        """Creates the coupling Dirichlet boundary conditions using
-        create_coupling_boundary_condition() method.
+            coupling_bc_expression = self._my_expression(element=function_space.ufl_element(), degree=0)
 
-        :return: dolfin.DirichletBC()
-        """
-        self._function_space = function_space
-        self._create_coupling_boundary_condition()
-        return dolfin.DirichletBC(self._function_space, self._coupling_bc_expression, self._coupling_subdomain)
+        coupling_bc_expression.set_boundary_data(self._read_data, x_vert, y_vert)
 
-    def create_coupling_neumann_boundary_condition(self, test_functions, function_space=None, boundary_marker=None):
-        """Creates the coupling Neumann boundary conditions using
-        create_coupling_boundary_condition() method.
+        return coupling_bc_expression
 
-        :return: expression in form of integral: g*v*ds. (see e.g. p. 83ff
-         Langtangen, Hans Petter, and Anders Logg. "Solving PDEs in Python The
-         FEniCS Tutorial Volume I." (2016).)
-        """
-        if not function_space:
-            self._function_space = test_functions.function_space()
-        else:
-            self._function_space = function_space
-        self._create_coupling_boundary_condition()
-        if not boundary_marker: # there is only 1 Neumann-BC which is at the coupling boundary -> integration over whole boundary
-            if self._coupling_bc_expression.is_scalar_valued():
-                return test_functions * self._coupling_bc_expression * dolfin.ds  # this term has to be added to weak form to add a Neumann BC (see e.g. p. 83ff Langtangen, Hans Petter, and Anders Logg. "Solving PDEs in Python The FEniCS Tutorial Volume I." (2016).)
-            elif self._coupling_bc_expression.is_vector_valued():
-                n = FacetNormal(self._mesh_fenics)
-                return -test_functions * dot(n, self._coupling_bc_expression) * dolfin.ds
-            else:
-                raise Exception("invalid!")
-        else: # For multiple Neumann BCs integration should only be performed over the respective domain.
-            # TODO: fix the problem here
-            raise Exception("Boundary markers are not implemented yet")
-            return dot(self._coupling_bc_expression, test_functions) * self.dss(boundary_marker)
-        
-    def create_force_boundary_condition(self, function_space):
+    def create_force_boundary_condition(self, Dirichlet_Boundary, function_space):
```

Comment:
I think we should also try to keep this interface a bit more abstract. Directly return the point sources to the user that are generated from data. This does not necessarily need to be a force. I can also take care of this, when I'm updating the FSI tutorials (see https://github.com/precice/tutorials/pull/62#pullrequestreview-356823828).

Reply:
I agree that more abstraction is necessary here, especially such that finally the mid-level interface should be able to solve CHT, HT and FSI cases with no special implementation. Currently I have concentrated on getting https://github.com/precice/tutorials/pull/62 running which is a CHT tutorial. Once this is running I plan to incorporate the FSI tutorial with your help. I hope this workflow is okay.